packages/react/src/search/Search.tsx

```diff
@@ -0,0 +1,58 @@
+import { useComposedRefs } from "@radix-ui/react-compose-refs";
+import { useControllableState } from "@radix-ui/react-use-controllable-state";
+import { type ComponentPropsWithRef, forwardRef, useRef } from "react";
+
+import type { ExtendProps } from "../utils";
+
+import { Box } from "../box";
+import { Button } from "../button";
+import { Input } from "../input";
+import { IconCross } from "./IconCross";
+import { IconSearch } from "./IconSearch";
+
+type SearchProps = ExtendProps<
+  ComponentPropsWithRef<typeof Box>,
+  Omit<ComponentPropsWithRef<typeof Input>, "onChange">,
+  {
+    defaultValue?: string;
+    onChange?: (value: string) => void;
+    value?: string;
+  }
+>;
+
+export const Search = forwardRef<HTMLInputElement, SearchProps>(
+  ({ defaultValue = "", onChange, value: valueProp, ...props }, outerRef) => {
+    const innerRef = useRef<HTMLInputElement>(null);
+    const ref = useComposedRefs(innerRef, outerRef);
+
+    const [value, setValue] = useControllableState({
+      defaultProp: defaultValue,
+      onChange: onChange,
+      prop: valueProp,
+    });
+
+    return (
+      <Input
+        endDecorator={
+          value && (
+            <Button
+              appearance="secondary"
+              icon={<IconCross />}
+              onClick={() => setValue("")}
```

Comment:
here we need to do a little trickery

https://stackoverflow.com/a/46012210

```suggestion
              onClick={() => {
                if (!innerRef.current) {
                  return;
                }

                Object.getOwnPropertyDescriptor(
                  innerRef.current.constructor.prototype,
                  "value",
                )?.set?.call(innerRef.current, "");
                innerRef.current?.dispatchEvent(
                  new Event("change", { bubbles: true }),
                );

                innerRef.current.focus();
              }}
```

don't forget to re-focus the input after we clear the value

Reply:
we basically re-trigger react's own `onChange` handler on the input - which allows us to set the value or propagate it up to the parent component